workers/unity/Assets/Playground/Scripts/Worker/AndroidClientWorkerConnector.cs

```diff
@@ -0,0 +1,60 @@
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Mobile;
+#if UNITY_ANDROID
+using Improbable.Gdk.Mobile.Android;
+#endif
+using System;
+using UnityEngine;
+
+
+namespace Playground
+{
+    public class AndroidClientWorkerConnector : MobileWorkerConnector, IMobileClient
+    {
+        public string IpAddress { get; set; }
+        public ConnectionScreenController ConnectionScreenController { get; set; }
+
+        [SerializeField] private GameObject level;
+
+        private GameObject levelInstance;
+
+        public async void TryConnect()
+        {
+            await Connect(WorkerUtils.AndroidClient, new ForwardingDispatcher()).ConfigureAwait(false);
+        }
+
+        protected override void HandleWorkerConnectionEstablished()
+        {
+            ConnectionScreenController.OnSuccess();
+
+            WorkerUtils.AddClientSystems(Worker.World);
+            if (level == null)
+            {
+                return;
+            }
+
+            levelInstance = Instantiate(level, transform);
+            levelInstance.transform.SetParent(null);
+        }
+
+        protected override void HandleWorkerConnectionFailure()
+        {
+            ConnectionScreenController.OnConnectionFailed();
+        }
+
+        protected override string GetHostIp()
+        {
+#if UNITY_ANDROID
+            if (Application.isMobilePlatform && DeviceInfo.IsAndroidStudioEmulator() && IpAddress.Equals(string.Empty))
+            {
+                return DeviceInfo.AndroidStudioEmulatorDefaultCallbackIp;
+            }
+
+            return IpAddress;
+#else
+            throw new PlatformNotSupportedException(
+                "This method is only defined for the Android platform. Please check your build settings.");
```

Comment:
This error may be confusing to users and may not really help them, I'd try to see what's the last user action that may trigger this, e.g. "You can use an AndroidWorkerConnector only for the Android Platform".

Additionally, perhaps it'd be better to throw something within OnEnable (or probably better, TryConnect?) if this behaviour is added when not UNITY_ANDROID?